src/dmd/expression.d

```diff
@@ -1871,6 +1871,25 @@ extern (C++) final class IntegerExp : Expression
         }
         return result;
     }
+
+    /**
+     * Use this instead of creating new instances for commonly used values such
+     * as 0 or 1 and when the expression value is never modified.
+     *
+     * Note that the constness cannot be guaranteed.
```

Comment:
Looks like an accident waiting to happen.  Maybe add an `assert(theConstant.getValue() == v)`.

Reply:
There's probably a way to guarantee constness, but you'd probably have to modify the inheritance hierarchy and the interfaces where it is used.

Reply:
The value is actually modifiable just because of the normalization. The method allowing to change the value is never used otherwise.